### ISSUE TITLE
any idea how to test with pytest and django_tenants ?

### ISSUE THREAD

**Issue by m7salam** [django-tenants/django-tenants#633]
I have been trying to use pytest with my project which is built with django tenants

but everytime running the tests I'm getting Table doesn't exist which is weird and I'm using the TenantTestCase that is provided by the library in the documentation 

```
@pytest.mark.django_db
class TestConvertableNote(TenantTestCase):
    @classmethod
    def setup_tenant(cls, tenant):
        """
        Add any additional setting to the tenant before it get saved. This is required if you have
        required fields.
        """
        tenant.username = tenant.schema_name
        return tenant

    def setUp(self):
        super().setUp()
        self.c = TenantClient(self.tenant)

    def test_convertable_loan_create(self):
        convertable_loan = ConvertableLoanFactory()
        assert isinstance(convertable_loan, ConvertableLoan)

    def test_convertable_loan_str(self):
        convertable_loan = ConvertableLoanFactory()
        assert convertable_loan.__str__() == convertable_loan.title

```

this is the factories.py file

```
class ConvertableLoanFactory(DjangoModelFactory):
    name = Faker("company")
    lender = SubFactory(TenantShareholderFactory, name="shareholder 1")

    class Meta:
        model = ConvertableLoan

```

any idea to know how to work smoothly with django tenants and doing unit tests using pytest, factory_boy and faker

thanks

**Comment by lothemar** [comment]
Had a similar issue, did you run migrations on the pytest db? Add the ```--create-db``` flag to ensure pytest isn't reusing the same (unmigrated) database.

**Comment by m7salam** [comment]
> Had a similar issue, did you run migrations on the pytest db? Add the `--create-db` flag to ensure pytest isn't reusing the same (unmigrated) database.

Yeah I did both 

it wasn't related 

at the end I have to give up using this class, I just created my own setup similar to how I use It and ran the tests through it by using BaseTest setup class that I created

but the above one I don't know why it was happening

**Comment by lothemar** [comment]
just to make sure: the problem was in TenantTestCase itself? If you could provide any relevant logs/any info, I'll investigate it more & open a separate issue

**Comment by m7salam** [comment]
@lothemar  that was the error coming and I posted the classes 

I just wanted to know if there is a certain way to deal with pytest and run test cases with django-tenants

Cuz the documentation not covering that 

So if you can give example with how would you handle the test cases with pytest would be very helpful as in steps that works for U 

Thanks

**Comment by lucasrcezimbra** [comment]
I did this way and worked:

```python
# conftest.py
from django_tenants.test.cases import TenantTestCase


@pytest.fixture(scope='session')
def django_db_setup(django_db_blocker):
    with django_db_blocker.unblock():
        TenantTestCase.setUpClass()

    yield

    with django_db_blocker.unblock():
        TenantTestCase.tearDownClass()
```

**Comment by tomturner** [comment]
Consolidating the backlog: this is the same underlying request as #664, which is where using pytest with django-tenants is being tracked. Closing here so the discussion stays in one place — please add anything from your case over there, and thanks for reporting it.
